### ISSUE TITLE
Add postfix to dependency for email sending

### ISSUE THREAD

**Issue by lieryan** [atomic-penguin/cookbook-gitlab#91]
After cookbook install, Gitlab cannot send emails (e.g. for new account creation, etc). Adding dependency to postfix would fix that.

**Comment by atomic-penguin** [comment]
Its necessary to have an LSB compatible MTA with a sendmail command for e-mail alerts from the application.  It is not necessary that MTA be postfix.

**Comment by lieryan** [comment]
That's true. But is there any reason the cookbook shouldn't install a default MTA if one isn't already provisioned in the machine (like how currently the cookbook installs MySQL if neither gitlab::mysql nor gitlab::postgres specified in the runlist)?

At the very least, I think it should be documented in the README that the user also need to either add an LSB-compatible MTA cookbook or add their own recipe to configure initializers/smtp_settings.rb to get email working. Since many important functionalities of Gitlab requires email, having it work straight from the oven would be nice.

**Comment by atomic-penguin** [comment]
The mysql issue was not our design decision.  It was more of a quick reaction to the removal of mysql::server recipe in the Chef maintained mysql cookbook.  Most likely will adopt an omnibus-gitlab installation and just drop mysql support in the 8.0 release of the cookbook as postgres will be included in the omnibus package.

I agree there needs to be an smtp_settings template to configure an SMTP connector.
